### ISSUE TITLE
Command line parsing should correct non-word branch names

### ISSUE THREAD

**Issue by acidprime** [puppetlabs/r10k#281]
r10k appears to currently silently fail if you pass it a branch name that has not been normalized. This likely should follow the same code path of normalization non-word

tested on r10k v1.3.1 , related to zendesk 9014

``` shell
[root@master5 environments]# pwd 
/etc/puppetlabs/puppet/environments 
[root@master5 environments]# r10k deploy environment example-branch -p 
[R10K::Source::Git - WARN] Environment "example-branch" contained non-word characters, correcting name to example_branch 
[R10K::Source::Git - WARN] Environment "example2-branch" contained non-word characters, correcting name to example2_branch 
[root@master5 environments]# ls example_branch 
ls: cannot access example_branch: No such file or directory 
[root@master5 environments]# r10k deploy environment example_branch -p 
[R10K::Source::Git - WARN] Environment "example-branch" contained non-word characters, correcting name to example_branch 
[R10K::Source::Git - WARN] Environment "example2-branch" contained non-word characters, correcting name to example2_branch 
[root@master5 environments]# ls example_branch 
correctstuff.sh
```

**Comment by adrienthebo** [comment]
Duplicate of #260 and should be released on 1.4.1, which I should probably do in the next few minutes. :)

**Comment by acidprime** [comment]
:heart:
